### DISCUSSION TITLE
Projecting GNSS Time Series into LOS

### DISCUSSION THREAD

**Discussion by Congzhangg** [gmtsar/gmtsar#467] [Q&A]
Hello everyone, At this step, I have a few questions I would like to ask you:

1.  how to use the  individual GNSS stations’ displacement time series to get an LOS  displacement time series for each station, The command is  :  paste GNSS_ts.enu GNSS.llhenu | awk '{print ($1*&7) + ($2*$8) + ($3+$9)}' > GNSS_ts.los  ?   
Or :    awk '{print ($1*elook)+($2*nlook)+($3*ulook) }' GNSS_ts.enu > GNSS_ts.los
![image](https://user-images.githubusercontent.com/112800803/195010805-6fa52811-2671-4d40-91da-970fe949a9f9.png)

2 . Extract Point Displacements from InSAR LOS Grids.   I have projected dis_*.grd to latitude, longitude coordinates, How to get the displacement time series of G100 station from all disp_*.grd files? 
The only information I consulted was the location of a dis_*.grd file from the radar.  just like : gmt grdtrack GNSS.rahll -G../SBAS/disp_2018031.grd > LOS_InSAR.disp
![image](https://user-images.githubusercontent.com/112800803/195014208-578a3b52-40db-4bb3-a223-ca1651b7c7f3.png)

**Comment by Xiaohua-Eric-Xu** [comment]
Your GNSS.ll looks strange to me as I don't remember a latitude that could be -221-ish.

**Reply by Xiaohua-Eric-Xu** [reply]
Saw your reply in email but could not find it here. Strange...

**Reply by Congzhangg** [reply]
> Saw your reply in email but could not find it here. Strange...

I have updated my Q & A ,i am looking forward for your reply!thanks!

**Reply by Xiaohua-Eric-Xu** [reply]
Please continue in the thread instead of updating the Q&A.

To extract data there is a script called extract_one_time_series.csh.

**Reply by Congzhangg** [reply]
Thank you ,I successfully extracted the time-series displacements of the points from InSAR LOS.  

However, I made an error when using a GNSS data to convert to LOS.  I have prepared the documents: G100_ts.enu 、GNSS.llhenu， Here are all the commands I tried, but none of them worked as well as I wanted
![image](https://user-images.githubusercontent.com/112800803/195781606-9f56a3f0-6a6a-49f3-bbe0-ab04e1be2812.png)

**Reply by Xiaohua-Eric-Xu** [reply]
You may have mistyped a $# in your awk command?

**Reply by Congzhangg** [reply]
I am sorry, Please tell me in more detail？The material gives two processing statements for the rate and the time series, the speed I know, but for the time series:   awk '{print ($1*elook)+ ($2*nlook)+($3*ulook)}' MAI_ts.enu > MAI_ts.los.    The command only entered a file called MAI ts.enu.  I'm not sure how to use the command to get the site time series displacement I want.

**Reply by Xiaohua-Eric-Xu** [reply]
![WX20221017-110101](https://user-images.githubusercontent.com/16844111/196226635-993771b8-925c-4299-ba85-9fc7e116ce13.png)

**Reply by Congzhangg** [reply]
OK, I know what you mean.  Why does this statement only get one line of information?
For a survey station, GNSS.enu contains many time series positions, but GNSS.llhenu has only one look vector about this survey station.

By the way, According to the documentation, why does the value of 'h' in GNSS.llhenu change after running the SAT_look command?

**Reply by Xiaohua-Eric-Xu** [reply]
I guess a station won't move a lot during your survey. Thus for Satellite look angle, the changes are negligible.

The changes in `h` is because it is converted from height above ellipsoid to above the reference radius in the PRM.

**Reply by Congzhangg** [reply]
When extract GNSS.enu file, The ENU value for the onset time is 0,0,0;  The displacement at each subsequent time is relative to the beginning of the study, right?

**Reply by Xiaohua-Eric-Xu** [reply]
I guess so.
